google/cloud/bigtable/read_rows_query.py

```diff
@@ -13,36 +13,158 @@
 # limitations under the License.
 #
 from __future__ import annotations
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any
+from .row_response import row_key
+from dataclasses import dataclass
+from google.cloud.bigtable.row_filters import RowFilter
 
 if TYPE_CHECKING:
-    from google.cloud.bigtable.row_filters import RowFilter
     from google.cloud.bigtable import RowKeySamples
 
 
+@dataclass
+class _RangePoint:
+    # model class for a point in a row range
+    key: row_key
+    is_inclusive: bool
+
+
 class ReadRowsQuery:
     """
     Class to encapsulate details of a read row request
     """
 
     def __init__(
-        self, row_keys: list[str | bytes] | str | bytes | None = None, limit=None
+        self,
+        row_keys: list[str | bytes] | str | bytes | None = None,
+        limit: int | None = None,
+        row_filter: RowFilter | dict[str, Any] | None = None,
```

Comment:
__init__ doesn't have row_ranges. is this intended?

Reply:
Yes, because this implementation doesn't have an explicit range object, so passing in multiple ranges at init time would be difficult. Users can call `Query().add_range(start, end)` to add ranges

I think Python's style is biased towards flattness, and avoiding custom objects when you can get away with it. But I'm not opposed to adding a range class if you expect users to be re-using or passing ranges around a lot